src/flags.ts

```diff
@@ -102,64 +72,75 @@ export function isPrecedingCharStringDelimeter(document: vscode.TextDocument, po
 
 const candidateTextStartLocation = (char: number) => (char === 1 ? 0 : char - 2);
 
-interface IFlagManager {
-	store: LDFeatureStore;
-	updateProcessor: LDStreamProcessor;
+async function openFlagInBrowser(key: string) {
+	const flag = await api.getFeatureFlag(config.project, key, config.env);
+
+	// Default to first environment
+	let env: Environment = Object.values(flag.environments)[0];
+	let sitePath = env._site.href;
+
+	if (!config.env) {
+		window.showWarningMessage('[LaunchDarkly] env is not set. Falling back to first environment.');
+	} else if (!flag.environments[config.env]) {
+		window.showWarningMessage(
+			`[LaunchDarkly] Configured environment '${config.env}' has been deleted. Falling back to first environment.`,
+		);
+	} else {
+		env = flag.environments[config.env];
+		sitePath = env._site.href;
+	}
+	opn(url.resolve(config.baseUri, sitePath));
 }
 
-export class LDFlagManager implements IFlagManager {
-	store = InMemoryFeatureStore();
+class FlagStore {
+	store: LDFeatureStore;
 	updateProcessor: LDStreamProcessor;
-	private settings: IConfiguration;
-
-	constructor(ctx: vscode.ExtensionContext, settings: IConfiguration) {
-		this.settings = Object.assign({}, settings);
-		let config = this.config(settings);
-		if (!settings.sdkKey) {
-			console.warn('LaunchDarkly sdkKey is not set. Language support is unavailable.');
-			return;
-		}
 
-		this.updateProcessor = StreamProcessor(settings.sdkKey, config, Requestor(settings.sdkKey, config));
+	constructor() {
+		this.store = InMemoryFeatureStore();
 		this.start();
 	}
 
+	reload(e: ConfigurationChangeEvent) {
+		if (['sdkKey', 'baseUri', 'streamUri'].some(option => e.affectsConfiguration(`launchdarkly.${option}`))) {
+			this.stop();
+			this.start();
+		}
+	}
+
 	start() {
-		this.updateProcessor &&
-			this.updateProcessor.start(err => {
-				if (err) {
-					let errMsg;
-					if (err.message) {
-						errMsg = `Error retrieving feature flags: ${err.message}.`;
-					} else {
-						console.error(err);
-						errMsg = `Unexpected error retrieving flags.`;
-					}
-					vscode.window.showErrorMessage(`[LaunchDarkly] ${errMsg}`);
+		if (!config.sdkKey || !config.baseUri || !config.streamUri) {
+			console.warn('LaunchDarkly extension is not configured. Language support is unavailable.');
+			return;
+		}
```

Comment:
Moved this logic into `start`. It's slightly different in that we're checking for all the crucial values for streaming to work.